model/eventQuestionModel.js

```diff
@@ -9,13 +9,55 @@ const eventQuestionSchema = Mongoose.Schema({
         type: Mongoose.Types.ObjectId,
         required: true
     },
+    companyID: {
+        type: Mongoose.Types.ObjectId,
+        required: true
+    },
     content: {
         type: String,
-        required: true
+        required: true,
+        trim: true,
+        minlength: [1, 'Question content cannot be left blank']
     },
     responses: {
-        type: Array
+        type: [new Mongoose.Schema({
+            content: {
+                type: String,
+                required: true,
+                trim: true
+            },
+            user: { // a simplified version of the user is stored - maybe just store the ID, but then there will be an influx of queries whilst showing the Q&A section => DISCUSS
+                type: new Mongoose.Schema({
+                    id: {
+                        type: String,
```

Comment:
Should be ObjectId?

Reply:
@zirlman since it's a simple representation, a string for id will be enough.

Reply:
A simplified version of user is stored. As such I did not see a reason to store it as an ObjectId. If you think there is a reason to change it to one, it is simple enough to do, since mongoose automatically converts it from string to ObjectId.